### PR TITLE
Add `number_field` in `FormBuilder`

### PR DESCRIPTION
**Pull request by elias19r**
Based on PR https://github.com/BuoySoftware/anchor_view_components/pull/130

In this PR:
- Add `number_field` method to `FormBuilder`, wrapping it with the `InputComponent`
- Add previews for number fields

### CODE REVIEW

File: app/helpers/anchor/form_builder.rb [hunk 1]
```diff
@@ -72,6 +72,17 @@ def text_field(attribute, options = {})
...
+    def number_field(attribute, options = {})
+      render NumberFieldComponent.new(
+        form_builder: self,
+        attribute:,
+        starting_icon: options.delete(:starting_icon),
+        ending_icon: options.delete(:ending_icon)
+      ) do |component|
+        super attribute, component.options.merge(options)
+      end
+    end
```
Comment (elias19r) [inline app/helpers/anchor/form_builder.rb:84]: I need an input of type `number` and I noticed this is not currently implemented in Anchor, so I'm opening this PR looking for suggestions on how I should implement it.

Comment (elias19r) [reply app/helpers/anchor/form_builder.rb:84]: How do you feel about this approach:

- Should we implement a `NumberFieldComponent` that is currently just a copy of `TextFieldComponent`?
- Should `TextFieldComponent` be renamed to `InputComponent` and then we would use it for both `#text_field` and `#number_field`?

Comment (elias19r) [reply app/helpers/anchor/form_builder.rb]: From PR https://github.com/BuoySoftware/anchor_view_components/pull/130 , `TextFieldComponent` is being renamed to `InputComponet`

**Review by thoughtbot-summer — APPROVED** [review]

**Review by arzezak — APPROVED** [review]
Great! 👏 Thanks.

**Review by tysongach — COMMENTED** [review]
I apologize I’m just getting to this PR…

What’s the motivation for a `type=number` input?

I ask because they are troubled in a few way - there’s a great CSS-Tricks article about it: https://css-tricks.com/what-to-use-instead-of-number-inputs/

`type=number` inputs give you those awkward steppers on the right side, which usually aren’t useful. What’s our need/intention and would an `inputmode=numeric` be better suited?

**Comment by elias19r** [comment]
Hi @tysongach!

I need an input field for receiving dollar amounts on the [Payment Plan Config Full Page](https://app.asana.com/0/1203986647279434/1205478542205265/f). I didn't know about the issues `type="number"` presents, thank you for sharing that!

Looking at `inputmode="numeric"`, or `inputmode="decimal"`, it works as well. The only benefit I see with `type="number"` is that it automatically doesn't allow the input of non-numeric chars. I think I can work around that with a Stimulus controller.

**Comment by elias19r** [comment]
Let me know what you think! I can revert this, but keep the dollar icon :)

**Comment by tysongach** [comment]
This might be a documentation concern. There are valid uses for `type=number`, albeit rarely. Let me think on that…

In your case, I think you’re spot on about using `inputmode=decimal` - that, to me, aligns with what you’re describing. I doubt anyone would use the steppers from `type=number` to increment to their desired price point. So I’d consider that harmful to the UX.

> The only benefit I see with `type="number"` is that it automatically doesn't allow the input of non-numeric chars. I think I can work around that with a Stimulus controller.

This depends on the browser. Firefox, for example, does not do this. Personally, I tend to recommend server-side validation, as client-side introduces concerns around accessibility. I’m not sure I know enough about the problem around that though, and whether people are inputting non-number values. It might be wise to at least strip `$` from values somewhere, at least.

**Comment by tysongach** [comment]
Maybe we should have `money_input` or something, which scaffolds all that stuff like `inputmode=decimal` for you… 🤔

**Pull request merged**